File: datastruct/bak/util/LinkedListStack.go
package util

import(
	"fmt"
)

type linkedListStack struct {
	data *linkedList
}

func LinkedListStack() *linkedListStack {
	return &linkedListStack{
		data : LinkedList(),
	}
}

func (this *linkedListStack) GetSize() int {
	return this.data.GetSize()
}

func (this *linkedListStack) IsEmpty() bool {
	return this.data.IsEmpty()
}


func (this *linkedListStack) Push(e E) {
	this.data.AddFirst(e)
}

func (this *linkedListStack) Pop() E {
	return this.data.RemoveFirst()
}

func (this *linkedListStack) Peek() E {
	return this.data.GetFirst()
}

func (this *linkedListStack) ToString() {
	fmt.Print("Stack: top")
	this.data.ToString()
}

File: php-datastruct/LinkedList/LinkedListStack.php
<?php

class LinkedListStack extends LinkedList
{
    public function __construct($e=null, $next=null)
    {
        parent::__construct();
    }

    public function getSize()
    {
        return $this->size;
    }

    public function isEmpty()
    {
        return $this->size == 0;
    }

    public function push($element)
    {
        $this->addFirst($element);
    }

    public function pop()
    {
        return $this->removeFirst();
    }

    public function peek()
    {
        return $this->getFrist();
    }

    public function toString()
    {
        $str = "LinkedListStack: top ";
        $str .= parent::toString();
        return $str;
    }
}